### PR TITLE
[WIP] Folding in audioset

### PR DESCRIPTION
**Pull request by ejhumphrey**
Will address #2. 

paging @bmcfee for early feedback ... I've made a couple executive decisions that I'd be curious to get a 👍 / 👎 down on, specifically:
- In getting rid of the `librosa` dependency, I naively copied out `normalize` and `tiny` from `librosa.util` without thinking too hard.
- I've moved the `audioset` functionality into a submodule on `openmic`, e.g. `openmic.audioset.vggish_input`, etc. This feels like a deeper hierarchy than needed _right now_, but: calling the top-level package `audioset` doesn't seem right, and neither does unpacking the `audioset` submodules into `openmic`. Maybe this is a premature optimization, but it seems reasonable to expect that other helper tools will live alongside `openmic.audioset`. Alternatively, we could break things out into `openmic.vggish.{input, slim, preprocess}`, which feels a little nicer (this gets my vote).
- I want to move the majority of `featurefy` functionality into `openmic`, but I've left it all in the script for now.
- on `util` submodules. There's some stuff in `openmic.audioset.util` that isn't audioset specific, and might warrant a `openmic.util`, but not _all_ of it (I think). Is having multiple `util` sub(sub) modules floating around a package bad practice?
- I'm planning to chain this up to travis / coveralls as a separate issue _after_ this is back to master, so that there's a meaningful initial state on code coverage. Agree / disagree?

Meanwhile, a couple todos for myself while I'm looking at this:
- [x] get checksums for the vggish model and compare post-download. Those are hosted on a Google-controlled CDN and while they hopefully won't change ... there's nothing we can do about that.
- [x] move `featurefy.load_input` to `vggish_input.soundfile_to_examples`
- [x] update the download-deps script for the openmic/audioset package structure
- [x] refactor `featurefy.main` to `openmic.vggish.transform()`, which opens TF contexts if none are provided.

### CODE REVIEW

**Comment by ejhumphrey** [comment]
alrighty, sorry this is kinda large, but it makes the VGGish API much simpler (see `scripts/featurefy.py`). I think it's review ready?

File: openmic/util.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import hashlib
```
Comment (bmcfee) [inline openmic/util.py]: should have a docstring here

File: openmic/vggish/model.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (bmcfee) [inline openmic/vggish/model.py]: +docstring

File: openmic/vggish/model.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import numpy as np
+
+from . import params, MODEL_PARAMS
```
Comment (bmcfee) [inline openmic/vggish/model.py]: is it kosher to .-import an all-caps/const?  seems strange to me

File: openmic/vggish/params.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+# limitations under the License.
+# ==============================================================================
+
+"""Global parameters for the VGGish model.
+
+See vggish_slim.py for more information.
+"""
+import os
+import pkg_resources
+from openmic.util import md5_file
```
Comment (bmcfee) [inline openmic/vggish/params.py]: should this be a relative import `..util`?

File: openmic/util.py (hunk 1 above)
Comment (ejhumphrey) [reply openmic/util.py]: sidebar from IRL: At least a one-line docstring of what this is for. encoding and shebang are optional but encouraged.

File: openmic/vggish/model.py (hunk 3 above)
Comment (ejhumphrey) [reply openmic/vggish/model.py]: resolved IRL, MODEL_PARAMS should be accessed as an attr of params

File: openmic/vggish/params.py (hunk 4 above)
Comment (ejhumphrey) [reply openmic/vggish/params.py]: yes, willfix

File: openmic/vggish/inputs.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    RuntimeError - Soundfile is unable to parse the file format.
+    """
+    examples = None
+    try:
+        y, sr = sf.read(filename, always_2d=True)
+        # Mono only, `waveform_to_examples` will take care of samplerate
+        y = y.mean(axis=-1)
+        examples = waveform_to_examples(normalize(y), sr)
+
+    except RuntimeError as derp:
```
Comment (ejhumphrey) [inline openmic/vggish/inputs.py]: IRL snapshot: get rid of `strict` nonsense, soundfile throws whatever exceptions it throws, we throw `ValueError` if loaded file is empty / garbage

**Comment by ejhumphrey** [comment]
took care of things, so i'ma mail this through so i can move on to the travis stuff

**Pull request merged**